### PR TITLE
[OS-921, OS-922] Build with conan

### PR DESCRIPTION
**Pull request by skarbat**
This PR builds the mercury repository for the RPi and generates a Debian Package.

### CODE REVIEW

**Comment by skarbat** [comment]
The package now builds fine on a chroot and both `mercury` and `mercury-dev` packages install correctly on the PI. Ready to be merged @tombettany .

File: debian/control [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+Build-Depends: debhelper(>= 9.0.0), build-essential, cmake, python-conan, lcov
+
+Package: mercury
+Architecture: any
+Depends: libc6, libstdc++6, libgcc1
+Description: Kano OS multiplatform API abstraction layer
+
+Package: mercury-dev
+Architecture: any
+Depends: mercury
```
Comment (tombettany) [inline debian/control]: Might want this dependency to require the same version of `mercury` as the one provided by this same package as the version of the headers is going to be heavily tied into the underlying version.

```suggestion
Depends: mercury  (= ${binary:Version})
```

File: debian/rules [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/make -f
+
+%:
+	dh $@
+
+override_dh_auto_test:
+	echo "FIXME: We need to fix gtest build for the RPi to run dh_auto_test"
```
Comment (tombettany) [inline debian/rules]: We might end up not even running the tests here as they will be tested elsewhere and up to now we haven't done this in this step but given how we want this to be entirely cross-platform, we might actually want to do this here.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -7,9 +7,12 @@ add_library(mercury SHARED
...
 )
-target_link_options(mercury
-	PRIVATE -coverage
-)
+
+# option below is available starting with cmake 3.13
+# cmake on the pi is v3.7.2
+#target_link_options(mercury
+#	PRIVATE -coverage
+#)
```
Comment (tombettany) [inline src/CMakeLists.txt]: If we are disabling `-coverage` as a link option, I'm surprised that we don't also need to disable the `target_compile_options` above for `-fprofile-arcs -ftest-coverage` as they will require linkage options.

File: debian/rules (hunk 2 above)
Comment (skarbat) [reply debian/rules]: I will leave the `echo` as it is then, meanwhile we clarify how to best progress

File: src/CMakeLists.txt (hunk 3 above)
Comment (skarbat) [reply src/CMakeLists.txt]: What I noticed is that `lcov` was trying to obtain information as we discussed on our pair-programming session, but it failed at some later stage for reason still can't grasp for lack of knowledge on this. Suggest we adress all of this testing on a later PR

File: debian/control (hunk 1 above)
Comment (skarbat) [reply debian/control]: That's a good point, thanks! fixed and building fine.

**Pull request merged**